.eslintrc.json

```diff
@@ -169,6 +169,12 @@
       "rules": {
         "@typescript-eslint/no-var-requires": ["off"]
       }
+    },
+    {
+      "files": ["src/layout/*/index.tsx"],
+      "rules": {
+        "react-hooks/rules-of-hooks": "off"
```

Comment:
It feels a little strange/painful to have to turn this rule off, but as long we are using classes I don't think this is an issue. It 
 might be a sign that we should implement our code differently, but I'm not sure. The rule is mainly used to ensure that hooks is initialized in top scope and for classes this rule does not make sense.

Reply:
Yup, I know.. 😆 Little on the fence on this one as well. I guess we could separate out the methods that are meant to be used as hooks, but I'm not sure I like that approach either.

Reply:
Sounds like a pair-programming issue to me, to discuss the different approaches that could be implemented. Anyway, this is not a task for this PR. 😃